plugins/modules/storage/pmem/pmem.py

```diff
@@ -0,0 +1,470 @@
+  community.general.pmem:
+    socket:
+      - id: 0
+        appdirect: 10
+        appdirect_interleaved: False
+        memorymode: 70
+        reserved: 20
+      - id: 1
+        appdirect: 10
+        memorymode: 80
+        reserved: 10
+'''
+
+import json
+import re
+import traceback
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    import xmltodict
+except ImportError:
+    HAS_XMLTODICT_LIBRARY = False
+    XMLTODICT_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_XMLTODICT_LIBRARY = True
+
+
+class PersistentMemory(object):
+    def __init__(self):
+        module = AnsibleModule(
+            argument_spec=dict(
+                appdirect=dict(type='int'),
+                appdirect_interleaved=dict(type='bool', default=True),
+                memorymode=dict(type='int'),
+                reserved=dict(type='int'),
+                socket=dict(
+                    type='list', elements='dict',
+                    options=dict(
+                        id=dict(required=True, type='int'),
+                        appdirect=dict(required=True, type='int'),
+                        appdirect_interleaved=dict(type='bool', default=True),
+                        memorymode=dict(required=True, type='int'),
+                        reserved=dict(type='int'),
+                    ),
+                ),
+            ),
+            required_together=(
+                ['appdirect', 'memorymode'],
+            ),
+            required_one_of=(
+                ['appdirect', 'memorymode', 'socket'],
+            ),
+            mutually_exclusive=(
+                ['appdirect', 'socket'],
+                ['memorymode', 'socket'],
+            ),
+        )
+
+        if not HAS_XMLTODICT_LIBRARY:
+            module.fail_json(
+                msg=missing_required_lib('xmltodict'),
+                exception=XMLTODICT_LIBRARY_IMPORT_ERROR)
+
+        self.ipmctl_exec = module.get_bin_path('ipmctl', True)
+        self.ndctl_exec = module.get_bin_path('ndctl', True)
+
+        self.appdirect = module.params['appdirect']
+        self.interleaved = module.params['appdirect_interleaved']
+        self.memmode = module.params['memorymode']
+        self.reserved = module.params['reserved']
+        self.socket = module.params['socket']
+
+        self.module = module
+        self.changed = False
+        self.result = []
+
+    def pmem_run_command(self, command, returnCheck=True):
+        self.module.log(msg="pmem_run_command: execute: %s" % command)
+
+        rc, out, err = self.module.run_command(command)
+
+        self.module.log(msg="pmem_run_command: result: %s" % out)
+
+        if returnCheck and rc != 0:
+            self.module.fail_json(msg="Error while running: %s" %
+                                  command, rc=rc, out=out, err=err)
+
+        return out
+
+    def pmem_run_ipmctl(self, command, returnCheck=True):
+
+        command = "%s " % self.ipmctl_exec + command
```

Comment:
Sorry, one thing I missed on my previous reviews: it's best to use the list form for `run_command`, and not the string form.
```suggestion
        command = [self.ipmctl_exec] + command
```
(`command` must be converted to a list as well where this function is called)